src/compute/src/compute_state.rs

```diff
@@ -907,7 +915,10 @@ impl ReportedFrontier {
 /// State maintained for a compute collection.
 pub struct CollectionState {
     /// Tracks the frontier that has been reported to the controller.
-    pub reported_frontier: ReportedFrontier,
+    ///
+    // TODO(teskje): Now that we explicitly track the collection's `as_of`, we might be able to
+    // simplify this to an `Antichain<Timestamp>` again.
+    reported_frontier: ReportedFrontier,
```

Comment:
Made private to make it more likely that the different paths that update the reported frontier also call `observe_snapshot_produced`. Unfortunately, most of these paths are in the same module and there is nothing but discipline keeping us from modifying `reported_frontier` directly.